uxarray/grid/validation.py

```diff
@@ -0,0 +1,44 @@
+import numpy as np
+from uxarray.io._ugrid import _is_ugrid
+from uxarray.constants import ERROR_TOLERANCE
+
+
+# validation helper functions
+def check_connectivity(self):
+    """Check if all nodes are referenced by at least one element.
+
+    If not, the mesh may have hanging nodes and may not a valid UGRID
+    mesh
+    """
+
+    # Check if all nodes are referenced by at least one element
```

Comment:
Should this check if all nodes are referenced by at least two elements? Wouldn't it still be an invalid UGRID mesh if there was just a single line connecting to it?

Reply:
> Should this check if all nodes are referenced by at least two elements? Wouldn't it still be an invalid UGRID mesh if there was just a single line connecting to it?

Sometimes there is ocean or land mesh, basically anything not fully covering.
There is a metric used in quality `degree,` the number of faces that a node is a part of, the min/max of this number can be reported in a future PR.